### PR TITLE
RHIROS-919 implement custom OS filter logic

### PR DESCRIPTION
**Pull request by r14chandra**
OS filter will show versions of systems that are registered with ROS. 

Screenshot from 2023-01-17 17-05-23

### CODE REVIEW

File: src/Routes/RosPage/RosPage.js [hunk 1]
```diff
@@ -178,11 +205,21 @@ class RosPage extends React.Component {
         });
     }
 
+    updateOsFilter = (value) => {
```
Comment (PreetiW) [inline src/Routes/RosPage/RosPage.js:208]: Nitpick: Can we keep the `OS` as capital her as we are following that in the component? 

```suggestion
    updateOSFilter = (value) => {
```

File: src/Routes/RosPage/RosPage.js [hunk 2]
```diff
@@ -345,6 +400,16 @@ class RosPage extends React.Component {
...
                                         }
+                                    },
+                                    {
+                                        label: OSFObject.label,
+                                        type: OSFObject.type,
+                                        value: `checkbox-os`,
+                                        filterValues: {
+                                            items: OSFObject.filterValues?.items,
+                                            onChange: (_e, values) => this.updateOsFilter(values),
+                                            value: osFilterValue
```
Comment (PreetiW) [inline src/Routes/RosPage/RosPage.js:411]: Same as [above comment](https://github.com/RedHatInsights/ros-frontend/pull/165/files#r1100249090)

File: src/Routes/RosPage/RosPage.js [hunk 3]
```diff
@@ -195,10 +232,21 @@ class RosPage extends React.Component {
             });
         }
 
+        if (deletedOSFilters.length > 0) {
+            const resetFiltersList = deletedOSFilters[0]?.chips.map((chip) =>{
```
Comment (PreetiW) [inline src/Routes/RosPage/RosPage.js:236]: Can we keep the name as `resetOSFilterList` as it is related to OS filters?

File: src/Routes/RosPage/RosPage.js [hunk 4]
```diff
@@ -247,15 +302,14 @@ class RosPage extends React.Component {
...
-
+            orderBy, orderDirection, disableExport, isColumnModalOpen, OSFObject } = this.state;
         return (
             this.props.showConfigSteps
                 ?   <ServiceNotConfigured/>
                 :   <Card className='pf-t-light  pf-m-opaque-100'>
                     <CardBody>
                         <ManageColumnsModal
-                            isModalOpen={this.state.isColumnModalOpen}
+                            isModalOpen={isColumnModalOpen}
```
Comment (PreetiW) [inline src/Routes/RosPage/RosPage.js:312]: Good work on extracting the state values form single object 👍🏻

File: src/Routes/RosPage/RosPage.js [hunk 5]
```diff
@@ -92,6 +95,30 @@ class RosPage extends React.Component {
...
+        osObject.filterValues = {};
+        this.fetchSystems({
+            perPage: -1
+        }).then((response) => {
+            osObject.filterValues.items = Array.from(new Set((response.data).reduce((filtered, system) => {
+                if (system.os) {
+                    filtered.push(system.os);
+                }
+
+                return filtered;
```
Comment (PreetiW) [inline src/Routes/RosPage/RosPage.js:111]: What you think about putting this part into a function where we extract OS versions from the response to improve readability?

**Review by PreetiW — CHANGES_REQUESTED** [review]
Thank you for the PR @r14chandra, just added minor comments have a look and then we are good to go! 
Also I have tested this PR against production and everything looks good 👍🏻

File: src/Routes/RosPage/RosPage.js (hunk 3 above)
Comment (r14chandra) [reply src/Routes/RosPage/RosPage.js:236]: On line [#L188](https://github.com/RedHatInsights/ros-frontend/blob/main/src/Routes/RosPage/RosPage.js#L188), we're using `resetFiltersList` for `state` filter as well and this constant is being used within its scope. I think we are good here.

File: src/Routes/RosPage/RosPage.js (hunk 5 above)
Comment (r14chandra) [reply src/Routes/RosPage/RosPage.js:111]: I extracted this part from `componentDidMount` function. This function mainly has logic related to the OS version. Extracting again would not make sense, if not, we can change the function name to `getOSversionObject` or something like that. What's your thought on this?

File: src/Routes/RosPage/RosPage.js (hunk 1 above)
Comment (r14chandra) [reply src/Routes/RosPage/RosPage.js]: Done!

File: src/Routes/RosPage/RosPage.js (hunk 2 above)
Comment (r14chandra) [reply src/Routes/RosPage/RosPage.js:411]: Done!

File: src/Routes/RosPage/RosPage.js [hunk 6]
```diff
@@ -195,10 +232,21 @@ class RosPage extends React.Component {
             });
         }
 
+        if (deletedOSFilters.length > 0) {
+            const resetFiltersList = deletedOSFilters[0]?.chips.map((chip) =>{
+                return chip?.name;
+            });
+            const activeOsFilters = this.state.osFilterValue.filter(filterName => !resetFiltersList.includes(filterName));
```
[the remark this answers is not in the document]
Comment (r14chandra) [reply src/Routes/RosPage/RosPage.js]: Done!

File: src/Routes/RosPage/RosPage.js (hunk 5 above)
Comment (PreetiW) [reply src/Routes/RosPage/RosPage.js:111]: okay let's keep it as it s for now, if needed we will refactor later 💯

**Review by PreetiW — APPROVED** [review]
Thank you for making the changes @r14chandra. The PR looks good 👍🏻 LGTM 💯

**Pull request merged**